app/controllers/player_sessions_controller.rb

```diff
@@ -0,0 +1,35 @@
+class PlayerSessionsController < ApplicationController
+  skip_before_action(:force_player_sign_in, { :only => [:new_session_form, :add_cookie] })
+
+  def new_session_form
+    render({ :template => "player_sessions/sign_in.html.erb" })
+  end
+
+  def add_cookie
+    player = Player.where({ :email => params.fetch(:email) }).at(0)
+    
+    the_supplied_password = params.fetch(:password)
+    
+    if player != nil
+      they_are_real = player.authenticate(the_supplied_password)
+    
+      if they_are_real == false
+        redirect_to("/sign_in", { :alert => "Password incorrect." })
+      else
+        session[:player_id] = player.id
```

Comment:
Similar decision here: to use `.store` or no? I'm okay with not using it if we don't want a dependency on our patches.

Reply:
I'll make the switch to `.fetch` and `.store`